zio-http/src/main/scala/zio/http/Response.scala

```diff
@@ -14,13 +14,15 @@ import zio.{Cause, Promise, Task, Trace, Unsafe, ZIO}
 
 import java.io.IOException
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
+import java.util.concurrent.atomic.AtomicReference
 
-final case class Response private (
+sealed abstract case class Response private (
   status: Status,
   headers: Headers,
   body: Body,
   private[zio] val attribute: Response.Attribute,
   private[zio] val httpError: Option[HttpError],
+  frozen: Boolean = false,
```

Comment:
We still have `attribute` and `httpError`, can't we eliminate them too?

Reply:
I have a subsequent branch started as a follow up to this one  in which I intend to address these.  I am afraid this one is starting to grow a bit bigger than I care for.

Reply:
`Response` has been refactored to address this.